gui/default/index.html

```diff
@@ -344,118 +344,190 @@ <h3 class="panel-title">
       </div>
 
       <!-- Login form -->
-      <div ng-if="!authenticated" class="center-block">
+      <div ng-if="!authenticated" class="col-md-6 col-md-offset-3">
         <h3 translate>Authentication Required</h3>
 
-        <form ng-submit="authenticatePassword()" class="col-md-6">
-          <div class="form-group">
-            <label for="user" translate>User</label>
-            <input id="user" class="form-control" type="text" name="user" ng-model="login.username" autofocus required autocomplete="username" />
-          </div>
+        <div class="panel-group" id="login-options">
 
-          <div class="form-group">
-            <label for="password" translate>Password</label>
-            <input id="password" class="form-control" type="password" name="password" ng-model="login.password" ng-trim="false" autocomplete="current-password" />
-          </div>
+          <div class="panel panel-default">
+            <button
+              class="btn panel-heading"
+              data-toggle="collapse"
+              data-parent="#login-options"
+              data-target="#login-option-password"
+              aria-expanded="{{isDefaultLoginOption('password') ? 'true' : 'false'}}"
+            >
```

Comment:
I emphatically disagree with that order of priorities, but ok, done in c393ab3ff131f3d64511cf8f5d4c9139548ab9b4. And to be fair I noticed that I did also line-break a couple of tags that didn't otherwise change (apart from indentation) - my bad, that didn't belong in this diff either way (or at the very least, it should have been its own commit).

> And it does add quite a bit of spaces, needlessly inflating the file size.

May I ask the reason for this concern? Because if you're worried about the size of the Git repo, I don't think you should be - Git by default compresses content in packfiles using zlib, and indentation compresses _very_ well (default `zlib-flate` settings compress this file from 80,513B to 11,316B). Likewise for keeping the file small over the network, I think one should make sure to use HTTP compression (which would make micro-optimizations like this wholly insignificant) before worrying about source file size.

Anyway, done in c393ab3ff131f3d64511cf8f5d4c9139548ab9b4.

Reply:
1. Again, I think HTTP compression would be the proper solution for that; [MDN claims](https://developer.mozilla.org/en-US/docs/Web/HTTP/Compression) that it's ubiquitously supported in servers and browsers.
2. I agree with the goal, but I'm of the opposite opinion that breaking long lines _does_ help diff-friendliness, and _a lot_ at that. But that is of course assuming that the line breaks were already there, before changing the substance, and as I said I admit and agree that I failed on that part here (and even worse as I introduced some line breaks with no change in substance).
3. Ah, I see. I hadn't considered that.
4. Yeah, on some level I agree, but I would still advocate for improving code style incrementally as and where changes are made, rather than condemning the entire file to forever remain in a style that harms readability and diff-friendliness. If the choice is between (1) never change style, (2) change all the style at once, and (3) change style incrementally with new changes, I think (3) is by far the best. (1) makes the code artificially hard to work with; (2) tends to be postponed forever, and if/when you do make the leap, it becomes a "history barrier" that makes change history very hard to track beyond that point (and also affects merges and rebases), in a way that (3) does not.

Anyway, sorry, I'm not really trying to convince you to change your mind, just wanted to share a different perspective. I do appreciate these comments and consider these "nitpicks" no less worthy than other comments.